src/utils/daos_hdlr.c

```diff
@@ -1869,46 +1872,415 @@ set_dm_args_default(struct dm_args *dm)
+	props->dpp_entries[10].dpe_type = DAOS_PROP_CO_ENCRYPT;
+	props->dpp_entries[11].dpe_type = DAOS_PROP_CO_OWNER;
+	props->dpp_entries[12].dpe_type = DAOS_PROP_CO_OWNER_GROUP;
+	props->dpp_entries[13].dpe_type = DAOS_PROP_CO_DEDUP;
+	props->dpp_entries[14].dpe_type = DAOS_PROP_CO_DEDUP_THRESHOLD;
+
+	/* Conditionally get the OID. Should always be true for serialization. */
+	if (get_oid) {
+		props->dpp_entries[prop_index].dpe_type = DAOS_PROP_CO_ALLOCED_OID;
+		prop_index++;
+	}
+
+	if (get_label) {
+		props->dpp_entries[prop_index].dpe_type = DAOS_PROP_CO_LABEL;
+		prop_index++;
 	}
 
+	if (get_roots) {
+		props->dpp_entries[prop_index].dpe_type = DAOS_PROP_CO_ROOTS;
+	}
+
+	/* Get all props except ACL first. */
+	rc = daos_cont_query(coh, NULL, props, NULL);
+	if (rc != 0) {
+		DH_PERROR_DER(ap, rc, "Failed to query container");
+		D_GOTO(out, rc);
+	}
+
+	/* Fetch the ACL separately in case user doesn't have access */
+	rc = daos_cont_get_acl(coh, &prop_acl, NULL);
+	if (rc == 0) {
+		/* ACL will be appended to the end */
+		props_merged = daos_prop_merge(props, prop_acl);
+		if (props_merged == NULL) {
+			DH_PERROR_DER(ap, rc, "Failed set container ACL");
+			D_GOTO(out, rc = -DER_INVAL);
+		}
+		daos_prop_free(props);
+		props = props_merged;
+	} else if (rc && rc != -DER_NO_PERM) {
+		DH_PERROR_DER(ap, rc, "Failed to query container ACL");
+		D_GOTO(out, rc);
+	}
+	rc = 0;
+	*_props = props;
+out:
+	daos_prop_free(prop_acl);
+	if (rc != 0)
+		daos_prop_free(props);
+	return rc;
+}
+
+/* check if cont status is unhealthy */
+static int
+dm_check_cont_status(struct cmd_args_s *ap, daos_handle_t coh, bool *status_healthy)
+{
+	daos_prop_t		*prop;
+	struct daos_prop_entry	*entry;
+	struct daos_co_status	stat = {0};
+	int			rc = 0;
+
+	prop = daos_prop_alloc(1);
+	if (prop == NULL)
+		return -DER_NOMEM;
+
+	prop->dpp_entries[0].dpe_type = DAOS_PROP_CO_STATUS;
+
 	rc = daos_cont_query(coh, NULL, prop, NULL);
 	if (rc) {
-		DH_PERROR_DER(ap, rc, "daos_cont_query() failed");
-		daos_prop_free(prop);
+		DH_PERROR_DER(ap, rc, "daos container query failed");
 		D_GOTO(out, rc);
 	}
 
-	for (i = 0; i < size; i++) {
-		dpe_vals[i] = prop->dpp_entries[i].dpe_val;
+	entry = &prop->dpp_entries[0];
+	daos_prop_val_2_co_status(entry->dpe_val, &stat);
+	if (stat.dcs_status == DAOS_PROP_CO_HEALTHY) {
+		*status_healthy = true;
+	} else {
+		*status_healthy = false;
 	}
-
+out:
 	daos_prop_free(prop);
+	return rc;
+}
+
+static int
+dm_serialize_cont_md(struct cmd_args_s *ap, struct dm_args *ca, daos_prop_t *props,
+		     char *preserve_props)
+{
+	int	rc = 0;
+	int	num_attrs = 0;
+	char	**names = NULL;
+	void	**buffers = NULL;
+	size_t	*sizes = NULL;
+	void	*handle;
+	int (*daos_cont_serialize_md)(char *, daos_prop_t *props, int, char **, char **, size_t *);
+
+	/* Get all user attributes if any exist */
+	rc = dm_cont_get_usr_attrs(ap, ca->src_coh, &num_attrs, &names, &buffers, &sizes);
+	if (rc != 0) {
+		rc = 1;
```

Comment:
This can't be right?  You're passing this value to a function that expect a daos error code later, would it be enough to simply remove this line?

Reply:
@ashleypittman thanks, yeah I missed this one.